lib/browser/api/dialog.js

```diff
@@ -70,147 +70,140 @@ const checkAppInitialized = function () {
+  }
 
-    let [window, options, callback] = parseArgs(...args)
+  return {settings, dialogProperties, wrappedCallback}
+}
+
+module.exports = {
+  showOpenDialog: function (args) {
+    let options = args
+    checkAppInitialized()
 
     if (options == null) {
       options = {
-        title: 'Save'
+        title: 'Open',
+        properties: ['openFile']
       }
     }
 
-    let {buttonLabel, defaultPath, filters, title, message, securityScopedBookmarks = false, nameFieldLabel, showsTagField} = options
+    const {settings, dialogProperties, wrappedCallback} = setupDialog('open', options)
 
-    if (title == null) {
-      title = ''
-    } else if (typeof title !== 'string') {
-      throw new TypeError('Title must be a string')
-    }
-
-    if (buttonLabel == null) {
-      buttonLabel = ''
-    } else if (typeof buttonLabel !== 'string') {
-      throw new TypeError('Button label must be a string')
-    }
-
-    if (defaultPath == null) {
-      defaultPath = ''
-    } else if (typeof defaultPath !== 'string') {
-      throw new TypeError('Default path must be a string')
-    }
-
-    if (filters == null) {
-      filters = []
-    }
+    settings.properties = dialogProperties
+    return binding.showOpenDialog(settings, wrappedCallback)
+  },
 
-    if (message == null) {
-      message = ''
-    } else if (typeof message !== 'string') {
-      throw new TypeError('Message must be a string')
-    }
+  showSaveDialog: function (args) {
+    let options = args
+    checkAppInitialized()
 
-    if (nameFieldLabel == null) {
-      nameFieldLabel = ''
-    } else if (typeof nameFieldLabel !== 'string') {
-      throw new TypeError('Name field label must be a string')
+    if (options == null) {
+      options = {
+        title: 'Save'
+      }
     }
 
-    if (showsTagField == null) {
-      showsTagField = true
-    }
+    const {settings, wrappedCallback} = setupDialog('save', options)
 
-    const wrappedCallback = typeof callback === 'function' ? function (success, result, bookmarkData) {
-      return success ? callback(result, bookmarkData) : callback()
-    } : null
-    const settings = {title, buttonLabel, defaultPath, filters, message, securityScopedBookmarks, nameFieldLabel, showsTagField, window}
     return binding.showSaveDialog(settings, wrappedCallback)
   },
 
-  showMessageBox: function (...args) {
+  showMessageBox: function (args) {
+    let options = args
     checkAppInitialized()
 
-    let [window, options, callback] = parseArgs(...args)
-
     if (options == null) {
       options = {
         type: 'none'
       }
     }
 
     let {
-      buttons, cancelId, checkboxLabel, checkboxChecked, defaultId, detail,
-      icon, message, title, type
+      window, buttons, cancelId, checkboxLabel, checkboxChecked, defaultId,
+      detail, icon, message, title, type, callback
     } = options
 
-    if (type == null) {
-      type = 'none'
-    }
+    if (type == null) type = 'none'
```

Comment:
I would argue against this style in general.
When you put several expressions or statements on the same line,
you make it more difficult to search through the git history to find authors or reasons of a certain change,
because when changing one expression you become an author of the whole line and thus of the all other statements.